ext/drx/scatter_gather_aarch64.c

```diff
@@ -48,8 +521,113 @@
 bool
 drx_expand_scatter_gather(void *drcontext, instrlist_t *bb, OUT bool *expanded)
 {
-    /* TODO i#3837: add support for AArch64. */
     if (expanded != NULL)
         *expanded = false;
-    return drmgr_current_bb_phase(drcontext) == DRMGR_PHASE_APP2APP;
+
+    if (drmgr_current_bb_phase(drcontext) != DRMGR_PHASE_APP2APP)
+        return false;
+
+    instr_t *sg_instr = NULL;
+    if (!scatter_gather_split_bb(drcontext, bb, &sg_instr)) {
+        return true;
```

Comment:
This seems backward: if it fails to split we return "successfully expanded"?  Or is the return value just confusing?  Consider renaming the function or reversing its return values so this reads more naturally.

Reply:
We aren't returning "successfully expanded". There is a separate `OUT bool *expanded` argument to indicate whether anything was expanded and when we return `true` here `expanded` has been set to `false`. I have added comments here and to `scatter_gather_split_bb()` that I hope clarify how this works